api/src/Entity/ScheduleEntry.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace App\Entity;
+
+use ApiPlatform\Core\Annotation\ApiResource;
+use Doctrine\Common\Collections\Criteria;
+use Doctrine\Common\Collections\Selectable;
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity
+ * @ORM\HasLifecycleCallbacks
+ */
+#[ApiResource]
+class ScheduleEntry extends BaseEntity implements BelongsToCampInterface {
+    /**
+     * @ORM\ManyToOne(targetEntity="Period", inversedBy="scheduleEntries")
+     * @ORM\JoinColumn(nullable=false, onDelete="cascade")
+     */
+    public ?Period $period = null;
+
+    /**
+     * @internal Do not set the {@see Activity} directly on the ScheduleEntry. Instead use {@see Activity::addScheduleEntry()}
+     * @ORM\ManyToOne(targetEntity="Activity", inversedBy="scheduleEntries")
+     * @ORM\JoinColumn(nullable=false, onDelete="cascade")
+     */
+    public ?Activity $activity = null;
+
+    /**
+     * @var int minutes since period start
+     * @ORM\Column(type="integer", nullable=false)
+     */
+    public int $periodOffset = 0;
+
+    /**
+     * @var int minutes
+     * @ORM\Column(type="integer", nullable=false)
+     */
+    public int $length = 0;
+
+    /**
+     * @ORM\Column(name="`left`", type="float", nullable=true)
+     * --> left is a MariaDB keyword, therefore escaping for column name necessary
+     */
+    public float $left = 0;
+
+    /**
+     * @ORM\Column(type="float", nullable=true)
+     */
+    public float $width = 1;
+
+    public function getCamp(): ?Camp {
+        return (null !== $this->period) ? $this->period->camp : null;
+    }
+
+    public function getCategory(): ?Category {
+        return (null !== $this->activity) ? $this->activity->category : null;
+    }
+
+    public function getNumberingStyle(): ?string {
+        $category = $this->getCategory();
+
+        return (null !== $category) ? $category->numberingStyle : null;
+    }
+
+    public function getColor(): ?string {
+        $category = $this->getCategory();
+
+        return (null !== $category) ? $category->color : null;
+    }
```

Comment:
Könnte man in PHP 8 mit dem [nullsafe operator](https://www.php.net/releases/8.0/en.php#nullsafe-operator) vereinfachen und `getCategory` löschen.
```php
public function getNumberingStyle(): ?string {
    return $this->activity?->category?->numberingStyle;
}

public function getColor(): ?string {
    return $this->activity?->category?->color;
}
```

Aber `getColor` wird sowieso nur in einem Test gebraucht, könnte man auch löschen. `getNumberingStyle` wird in `getScheduleEntryNumber` benutzt, aber das ist wieder eine Methode die ich eher in einen Service verpacken würde, und dann könnte dieser Service auch selber den Numbering Style via Activity und Category rausholen.